appservice/src/tree/DeploymentTreeItem.ts

```diff
@@ -0,0 +1,103 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import * as path from 'path';
+import { ProgressLocation, TextDocument, window, workspace } from 'vscode';
+import { AzureTreeItem } from 'vscode-azureextensionui';
+import KuduClient from 'vscode-azurekudu';
+import { DeployResult, LogEntry } from 'vscode-azurekudu/lib/models';
+import { ext } from '../extensionVariables';
+import { DeploymentsTreeItem } from './DeploymentsTreeItem';
+import { ISiteTreeRoot } from './ISiteTreeRoot';
+
+export class DeploymentTreeItem extends AzureTreeItem<ISiteTreeRoot> {
+    public static contextValue: string = 'deployment';
+    public readonly contextValue: string = DeploymentTreeItem.contextValue;
+    public label: string;
+    public active: boolean;
+    public receivedTime: Date;
+    public parent: DeploymentsTreeItem;
+    private _kuduClient: KuduClient;
+    private _deployResult: DeployResult;
+
+    constructor(parent: DeploymentsTreeItem, deployResult: DeployResult, kuduClient: KuduClient) {
+        super(parent);
+        this._deployResult = deployResult;
+        if (!this._deployResult.receivedTime || !this._deployResult.message || this._deployResult.active === undefined) {
+            throw new Error('Invalid Deployment Result.');
+        }
+        this._kuduClient = kuduClient;
```

Comment:
Don't store something like kuduClient. The credentials could've changed so you should create a new client each time